[Go] Write createfollowship openapi to separate handler for test isolation

Rwatana opened the pull request:
## Issue Number
#676 

## Implementation Summary
- This pull request implements the CreateFollowship handler.
- Separation of the handler (create_followship.go) for better test isolation
- Addition of the OpenAPI path definition (paths/followship.yml)
- Addition of request/response schema:
- components/requests/create_followship_request.yml
- components/responses/create_followship_response.yml
- Update to openapi.yml to register the new endpoint

## Scope of Impact

This will be executed when a user sends a POST /api/users/{id}/follow request with a valid JSON payload (target_user_id).

## Particular points to check

- Make sure the request/response structure in OpenAPI aligns with the actual handler logic
- No regression is introduced to the existing server.go or handler registration
- The new endpoint doesn’t conflict with existing user-related endpoints
- OpenAPI bundling and validation (make bundle-openapi) succeeds without errors
## Test
```
go test -v ./...
```
## Schedule
4/19

## Note
Task Instruction from Kawai
- Set openapi schema
- Create a new create_followship.go file
- Define the necessary struct(s) for the handler
- Copy and migrate the relevant function from the existing handlers to this new file
- Register the new handler in server.go

Rwatana commented:
@nayuta-ai 
## Summary
I have added the task background to the Background section of the issue.
Please review my code.

nayuta-ai commented on openapi/openapi.yml:
```diff
@@ -23,6 +23,8 @@ paths:
     $ref: ./paths/find_user_by_id.yml
   /api/login:
     $ref: ./paths/login.yml
+  /api/users/{id}/following:
```
I'd like to confirm if the specification here is correct, so could you share the API definition of X that you're using as a reference?

Rwatana replied:
I read [[this](https://github.com/okuda-seminar/X-Clone-Backend/blob/b5e2a75f74b683fbde86b2b098a1d07292db1764/cmd/main.go#L68)](https://github.com/okuda-seminar/X-Clone-Backend/blob/b5e2a75f74b683fbde86b2b098a1d07292db1764/cmd/main.go#L68)  when implementing this part.

nayuta-ai commented on go/internal/controller/handler/create_followship.go:
```diff
@@ -0,0 +1,52 @@
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&body)
+	if err != nil {
+		http.Error(w, fmt.Sprintln("Request body was invalid."), http.StatusBadRequest)
```
```suggestion
		http.Error(w, "Request body was invalid.", http.StatusBadRequest)
```

nayuta-ai commented on go/internal/controller/handler/create_followship.go:
```diff
@@ -0,0 +1,52 @@
+
+	err = h.usecase.FollowUser(sourceUserID, body.TargetUserID)
+	if err != nil {
+		http.Error(w, fmt.Sprintln("Could not create followship."), http.StatusInternalServerError)
```
```suggestion
		http.Error(w, "Could not create followship.", http.StatusInternalServerError)
```

Rwatana commented:
@nayuta-ai 
I’ve provided the reference link I used when defining the API endpoint, and I’ve also refined the error response message.
Please review my code.

nayuta-ai commented:
I wanted to check if it followed the actual API functions. I was asking because I didn't know whether the current X-Clone followed API v1 or v2. Do you understand that?

nayuta-ai commented on go/internal/controller/handler/create_followship.go:
```diff
@@ -0,0 +1,51 @@
+	"x-clone-backend/internal/infrastructure/persistence"
+)
+
+// CreateFollowshipHandler handles the POST /api/users/{id}/follow
```
nit: /api/users/{id}/following

Rwatana commented:
@nayuta-ai 
Thanks for the clarification! I hadn’t considered the differences between API v1 and v2 when implementing the endpoint.

I referred to [this documentation](https://docs.x.com/x-api/users/follow-user), and based on that, I believe my implementation is correct. What do you think?

nayuta-ai commented:
I believe the code is fine.
For final confirmation, please share a curl command or an equivalent test to verify the current implementation.

Amoonguses1 commented:
@Rwatana 
Delete `mux.HandleFunc` for registering `CreateFollowship` in `cmd/main.go`.
This is the first time I'm saying this, but the request/response should be camelCase.

Rwatana commented:
@nayuta-ai 
## Topic
I have addressed the points pointed out by Kawai-kun:

- Removed the description in main.go
- Fixed the response/request field to use camelCase

Please check my code.
## Testing results:

Success case
<img width="1308" alt="スクリーンショット 2025-04-17 20 48 55" src="https://github.com/user-attachments/assets/66ce78f9-9b7c-4827-9ab4-f588da863a61" />

Failure case (400 Bad Request)
<img width="1308" alt="スクリーンショット 2025-04-17 20 49 58" src="https://github.com/user-attachments/assets/fa46abc9-3ba3-49d5-875c-7ba3226ea251" />

Failure case (500 Internal Server Error)
<img width="1308" alt="スクリーンショット 2025-04-17 20 49 52" src="https://github.com/user-attachments/assets/a4351d4a-60dd-445d-aa2f-311e6ec15562" />

nayuta-ai reviewed (approved):
lgtm
